### ISSUE TITLE
cannot reproduce "with EE" (with embeddings) paper results using this code

### ISSUE THREAD

**Issue by mirekphd** [entron/entity-embedding-rossmann#20]
In case of XGBoost I could find in this repo any functions that would use embeddings. As I understood, entity embeddings are produced by [__build_keras_model()](https://github.com/entron/entity-embedding-rossmann/blob/ae6ffc3b60718b5744582f7e9a5733818b31d3b1/models.py#L196), which seems to be used only for deep learning here, even though the paper shows its results in Tables III and IV also for KNN, RF, and XGB.

Helping others to reproduce accuracy gains from the use of embeddings in XGBoost is important because XGBoost baseline accuracy is difficult to improve using any neural network-based method, not just embeddings.

**Comment by entron** [comment]
Hi @mirekphd, in the file `train_test_model.py` there are two parameters related to your questions:

```
embeddings_as_input = False
save_embeddings = True
```
With the default setting as above the code will use NN to generate embeddings and save the embedding in the file `embeddings.pickle`.

After you generated the embeddings you can run the code with the parameter `embeddings_as_input = True` to use the embeddings for KNN, RF, and XGB. Hope I have answered your questions.

**Comment by mirekphd** [comment]
OK, now I think I understand. The imported pickled embeddings which were produced as a byproduct of training the DNN (which has to be trained anyway - that was the unintuitive part) will be then appended (once we change `embeddings_as_input` from False to True on next run) to the features matrix `X` (using `embed_features()` invoked [here](https://github.com/entron/entity-embedding-rossmann/blob/master/train_test_model.py#L31)) and `X` with those extra features can then be used by any algorithm, including XGBoost, or even linear model. Is this correct?

**Comment by entron** [comment]
You got it right:)

**Comment by mirekphd** [comment]
Thank you! So Is suppose we can close the issue.
